src/promises/allNoLeaks.ts

```diff
@@ -0,0 +1,38 @@
+// From https://stackoverflow.com/questions/64928212/how-to-use-promise-allsettled-with-typescript#69451755
```

Comment:
IANAL but are we're allowed to use code copied from stack overflow (I'm not sure that CC 4.0 SA is compatible with AGPL)?

Can we just write these ourselves?  They seem pretty trivial, and the code in that answer is kind of weird anyway (specifically, the type guard feature of this function doesn't belong / isn't doing anything)

I think we can just do:

```
const isRejected = (result: PromiseSettledResult) => result.status === 'rejected';
const isFulfilled = (result: PromiseSettledResult) => result.status === 'fulfilled';
```

and drop attribution since that's what we would have written all by ourselves!

Reply:
Let me see about that improvement. I think/hope we can credit the SO answer for the approach without violating copyright.